### PR TITLE
Cleanup HeapReporter command line options

### PR DESCRIPTION
**Pull request by philass**
This PR uses `llvm::cl::list` to handle the parsing of a list of passes past as command line options. 

This is consistent with the `onnx-const-prop-disable-pattern` and `functions-to-decompose` implementations for list command line parsing.

### CODE REVIEW

File: src/Compiler/HeapReporter.cpp [hunk 1]
```diff
@@ -48,15 +39,23 @@ void logMessage(StringRef logFilename, StringRef msg,
...
 
-HeapReporter::HeapReporter(
-    std::string logFilename, StringRef beforePasses, StringRef afterPasses)
+HeapReporter::HeapReporter(std::string logFilename,
+    std::vector<std::string> beforePasses, std::vector<std::string> afterPasses)
     : logFilename(logFilename) {
-  splitToSet(beforePasses, this->beforePassesSet);
-  splitToSet(afterPasses, this->afterPassesSet);
+  beforePassesSet = llvm::StringSet<>(beforePasses);
+  afterPassesSet = llvm::StringSet<>(afterPasses);
```
Comment (sorenlassen) [inline src/Compiler/HeapReporter.cpp:46]: can you set before/afterPassesSet in the initialization list?
```
  : logFilename(logFilename), beforePassesSet(beforePasses), afterPassesSet(afterPasses) {}
```

File: src/Compiler/HeapReporter.cpp [hunk 2]
```diff
@@ -48,15 +39,23 @@ void logMessage(StringRef logFilename, StringRef msg,
...
+  for (std::string &s : beforePasses)
+    beforePassesStr = beforePassesStr + s + ",";
+  for (std::string &s : afterPasses)
+    afterPassesStr = afterPassesStr + s + ",";
+
   reportBegin("onnx-mlir heap report"
               "\n--report-heap-before='" +
-              beforePasses.str() + "'\n--report-heap-after='" +
-              afterPasses.str() + "'");
+              beforePassesStr + "'\n--report-heap-after=" + afterPassesStr);
```
Comment (sorenlassen) [inline src/Compiler/HeapReporter.cpp:58]: can you use llvm::join() from StringExtras.h?
```
llvm::join(beforePasses, ",") + "'\n--report-heap-after=" + llvm::join(afterPasses, ",")
```

**Review by sorenlassen — APPROVED** [review]

File: src/Compiler/HeapReporter.cpp (hunk 1 above)
Comment (philass) [reply src/Compiler/HeapReporter.cpp]: Good catch

File: src/Compiler/HeapReporter.cpp (hunk 2 above)
Comment (philass) [reply src/Compiler/HeapReporter.cpp]: I knew there must be some handy function for this lying around. Thanks for the pointer

**Pull request merged**